mixer/mixer.go

```diff
@@ -0,0 +1,309 @@
+// Copyright 2023 LiveKit, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mixer
+
+import (
+	"fmt"
+	"slices"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/frostbyte73/core"
+	msdk "github.com/livekit/media-sdk"
+
+	"github.com/livekit/media-sdk/internal/ringbuf"
```

Comment:
We could move it from internal now, I think. We didn't want anyone using these packages by importing SIP, but here it would be appropriate.